web/app/js/components/Sidebar.jsx

```diff
@@ -148,54 +204,31 @@ class Sidebar extends React.Component {
               </PrefixedLink>
             </Menu.Item>
 
-            <Menu.Item className="sidebar-menu-item" key="/namespaces">
-              <PrefixedLink to="/namespaces">
-                <Icon type="dashboard" />
-                <span>Namespaces</span>
-              </PrefixedLink>
-            </Menu.Item>
-
-            <Menu.Item className="sidebar-menu-item" key="/tap">
-              <PrefixedLink to="/tap">
-                <Icon type="filter" />
-                <span>Tap</span>
-              </PrefixedLink>
-            </Menu.Item>
-
             {
-              _.map(_.take(this.state.namespaces, this.state.maxNsItemsToShow), ns => {
+              _.map(_.keys(sidebarComponents).sort(), resourceName => {
                 return (
-                  <Menu.Item className="sidebar-submenu-item" key={`/namespaces/${ns}`}>
-                    <PrefixedLink to={`/namespaces/${ns}`}>
-                      <Icon>{this.state.collapsed ? _.take(ns, 2) : <span>&nbsp;&nbsp;</span> }</Icon>
-                      <span>{ns} {this.state.collapsed ? "namespace" : ""}</span>
-                    </PrefixedLink>
-                  </Menu.Item>
+                  <SubMenu
+                    className="sidebar-menu-item"
+                    key={resourceName}
+                    title={<span>{friendlyTitle(resourceName).plural}</span>}>
+                    {
+                      _.map(_.sortBy(sidebarComponents[resourceName], r => r.resource.name), ro => {
+                        return (
+                          <Menu.Item key={"/namespaces/" + ro.resource.namespace + "/" + ro.resource.type + "s/" + ro.resource.name}>
```

Comment:
we can increase readability by refactoring the `"/namespaces/" + ro.resource.namespace + "/" + ro.resource.type + "s/" + ro.resource.name` so we don't repeat this particular string so many times here